receiver/awscontainerinsightreceiver/internal/gpu/dcgmscraper.go

```diff
@@ -0,0 +1,235 @@
+				SourceLabels: model.LabelNames{"Hostname"},
+				TargetLabel:  "NodeName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"namespace"},
+				TargetLabel:  "Namespace",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			// hacky way to inject static values (clusterName & instanceId) to label set without additional processor
+			// relabel looks up an existing label then creates another label with given key (TargetLabel) and value (static)
+			{
+				SourceLabels: model.LabelNames{"namespace"},
+				TargetLabel:  "ClusterName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  opts.HostInfoProvider.GetClusterName(),
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"namespace"},
+				TargetLabel:  "InstanceId",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  opts.HostInfoProvider.GetInstanceID(),
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"pod"},
+				TargetLabel:  "FullPodName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"pod"},
+				TargetLabel:  "PodName",
+				Regex:        relabel.MustNewRegexp("(.+)-(.+)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			// additional k8s podname for service name decoration
+			{
+				SourceLabels: model.LabelNames{"pod"},
+				TargetLabel:  "K8sPodName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"container"},
+				TargetLabel:  "ContainerName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"device"},
+				TargetLabel:  "GpuDevice",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+		},
+	}
+
+	promConfig := prometheusreceiver.Config{
+		PrometheusConfig: &config.Config{
+			ScrapeConfigs: []*config.ScrapeConfig{scrapeConfig},
+		},
+	}
+
+	params := receiver.CreateSettings{
+		TelemetrySettings: opts.TelemetrySettings,
+	}
+
+	promFactory := prometheusreceiver.NewFactory()
+	promReceiver, err := promFactory.CreateMetricsReceiver(opts.Ctx, params, &promConfig, opts.Consumer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create prometheus receiver: %w", err)
+	}
+
+	return &DcgmScraper{
+		ctx:                opts.Ctx,
+		settings:           opts.TelemetrySettings,
+		host:               opts.Host,
+		hostInfoProvider:   opts.HostInfoProvider,
+		prometheusReceiver: promReceiver,
+	}, nil
+}
+
+func (ds *DcgmScraper) GetMetrics() []pmetric.Metrics {
```

Comment:
I'm guessing we're doing this to follow some interface. Is that right?

Reply:
Looking at it again, it doesn't look like it's being used as part of an interface.